h2o-genmodel/src/main/java/hex/genmodel/algos/coxph/CoxPHMojoReader.java

```diff
@@ -0,0 +1,81 @@
+package hex.genmodel.algos.coxph;
+
+import hex.genmodel.ModelMojoReader;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class CoxPHMojoReader extends ModelMojoReader<CoxPHMojoModel> {
+
+  @Override
+  public String getModelName() {
+    return "CoxPH";
+  }
+
+  @Override
+  protected void readModelData() throws IOException {
+    _model._x_mean_cat = readkv2D_doubles("x_mean_cat");
+    _model._x_mean_num = readkv2D_doubles("x_mean_num");
+    _model._coef = readkv("coef");
+    _model._coef_names = readCoefNames();
+    _model._strata = readStrata();
+    _model._strata_len = readStrataLen();
+  }
+
+  private String[] readCoefNames() {
+    final int count = readkv("coef_names_num");
+    final String[] result = new String[count];
+    for (int i = 0; i < count; i++) {
+      result[i] = readkv("coef_names_" + i);
+    }
+    return result;
+  }
+
+  private Map<List<Integer>, Integer> readStrata() {
+    final int count = readkv("strata_count");
+    
+    final Map<List<Integer>, Integer> result = new HashMap<List<Integer>, Integer>(count);
+
+    for (int i = 0; i < count; i++) {
+      final double[] strata = readkv("strata_" + i);
+      final List<Integer> strataAsList = new ArrayList<>(strata.length);
+      for (int j = 0; j < strata.length; j++) {
+        strataAsList.add(j, (int) strata[j]);
+      }
+      result.put(strataAsList, i);
+    }
+    return result;
+  }
+  
+  private int readStrataLen() {
+    final int count = readkv("strata_count");
+    
+    if (0 == count) {
+      return 0;
+    } else {
+      final double[] strata = readkv("strata_0");
+      return strata.length;
+    }
+  }
+
+  private double[][] readkv2D_doubles(String keyPrefix) {
```

Comment:
probably also already implemented

Reply:
It looks like a funny story... it is probably implemented as `read2DArray` for GAM only :-/
So `writeDoubleArray` exists for all the writers but you can't use it anyway unsles you write your own code (that needs sizes of the array).

Reply:
it should be refactored to be reusable if possible

Reply:
Some of the code reused and moved to a more generic `ModelMojoReader`.